packages/react/src/hooks/useLauncher.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  getTreasureLauncherAuthToken,
+  startUserSessionViaLauncher,
+} from "@treasure-dev/launcher";
+import type { SessionOptions, TDKAPI, User } from "@treasure-dev/tdk-core";
+import { type ReactNode, useCallback, useEffect } from "react";
+import { AccountModal } from "../components/launcher/AccountModal";
+import { setStoredAuthToken } from "../utils/store";
+
+type Props = {
+  getAuthTokenOverride?: () => string | undefined;
+  tdk: TDKAPI;
+  setRootElement: (el: ReactNode) => void;
+  setUser: (user: User) => void;
+  onConnect?: (user: User) => void;
+};
+
+export const useLauncher = ({
+  getAuthTokenOverride,
+  tdk,
+  setUser,
+  setRootElement,
+  onConnect,
+}: Props) => {
+  const authToken = getAuthTokenOverride?.() ?? getTreasureLauncherAuthToken();
+  const isUsingTreasureLauncher = authToken !== undefined;
+
+  const startUserSessionViaLauncherIfNeeded = useCallback(
```

Comment:
It doesn't look like we need to memoize this. Also I think it's safe to remove the `IfNeeded` part off the name, it's a bit verbose